### PR TITLE
Add initial support removing SystemC

### PR DESCRIPTION
**Pull request by johnjohnlin**

### CODE REVIEW

**Comment by yodalee** [comment]
I think you are right. Keep the old version for now.

**Comment by yodalee** [comment]
Link #132 to track the issue.

File: vtuber/modellib/cfifo.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	void write(const T& t) {
+		DCHECK(not full());
+		--num_can_write_;
+		buf_.push_back(t);
+	}
+	void write(T&& t) {
+		DCHECK(not full());
+		--num_can_write_;
+		buf_.push_back(std::move(t));
+	}
```
Comment (yodalee) [inline vtuber/modellib/cfifo.h:44]: If I understand right, the read/write will cause error or UB if you try to read when empty or write when full. Right?

Comment (johnjohnlin) [reply vtuber/modellib/cfifo.h:44]: Yes, that is the purpose of DCHECK here.

**Comment by yodalee** [comment]
No further issue. Let's try in dev branch.

**Pull request merged**